### DISCUSSION TITLE
posts/xml-read-benchmarks/

### DISCUSSION THREAD

**Discussion by giscus** [BYJRK/byjrk.github.io#19] [Announcements]
# posts/xml-read-benchmarks/

最近工作中遇到了读写 XML 的情形，所以索性简单探索了一下相关的性能调优，在这里把我的发现分享给大家。

https://blog.coldwind.top/posts/xml-read-benchmarks/

**Comment by lgqlgqcool** [comment]
博主，在使用Xml.linq时，对于以下代码
var doc = XDocument.Parse(testXml);
    return doc
        .Elements("food")
        .Select(node => node.Element("name")!.Value)
        .ToList();
我实测是失败的，必须要从根节点逐层获取节点才可以正确获取到name节点

**Reply by BYJRK** [reply]
感谢指出。少了一个对于 `Root` 节点（也就是 `breakfast_menu`）的访问
